app/views/teachers/teacher-dashboard/BaseResourceHub/index.vue

```diff
@@ -0,0 +1,277 @@
+  import ButtonResourceIcon from './components/ButtonResourceIcon'
+  import { getResourceHubResources, getResourceHubZendeskResources } from 'core/api/resource_hub_resource'
+  import utils from 'app/core/utils'
+  const store = require('core/store')
+
+  const resourceSortFn = (a, b) => {
+    if (a.priority > b.priority) return 1  // Resource Hub Resources should usually have priorities
+    if (a.priority < b.priority) return -1
+    if (a.section_id > b.section_id) return 1  // Zendesk articles have section_ids
+    if (a.section_id < b.section_id) return -1
+    if (!a.promoted && b.promoted) return 1  // Zendesk articles can be promoted within their sections
+    if (a.promoted && !b.promoted) return -1
+    if (a.position > b.position) return 1  // Zendesk articles can have positions
+    if (a.position < b.position) return -1
+    if (a.name > b.name) return 1
+    if (a.name < b.name) return -1
+    return 0
+  }
+
+  const resourceHubSections = [
+    { sectionName: 'gettingStarted', slug: 'getting-started', i18nKey: 'teacher.getting_started' },
+    { sectionName: 'educatorResources', slug: 'educator-resources', i18nKey: 'new_home.educator_resources' },
+    { sectionName: 'studentResources', slug: 'student-resources', i18nKey: 'teacher.student_resources' },
+    { sectionName: 'lessonSlides', slug: 'lesson-slides', i18nKey: 'teacher.lesson_slides' },
+    { sectionName: 'faq', slug: 'faq', i18nKey: 'nav.faq' },
+  ]
+
+  export default {
+    name: 'BaseResourceHub',
+    components: {
+      ButtonResourceIcon,
+    },
+
+    data () {
+      return {
+        resourceHubResources: {},
+      }
+    },
+
+    computed: {
+      ...mapGetters({
+        loading: 'teacherDashboard/getLoadingState',
+        activeClassrooms: 'teacherDashboard/getActiveClassrooms'
+      }),
+
+      resourceHubSections () {
+        return resourceHubSections
+      },
+
+      resourceHubLinks () {
+        return (sectionName) => Object.values(this.resourceHubResources).filter((r) => r.section === sectionName).sort(resourceSortFn)
+      }
+    },
+
+    mounted () {
+      this.setTeacherId(me.get('_id'))
+      this.setPageTitle('Resource Hub')
+      this.fetchData({ componentName: this.$options.name, options: { loadedEventName: 'Resource Hub: Loaded' } })
+
+      getResourceHubResources().then(allResources => {
+        if (!Array.isArray(allResources) || allResources.length === 0) {
+          return
+        }
+
+        for (const resource of allResources) {
+          if (resource.hidden === true) {
+            continue
+          }
+
+          resource.name = utils.i18n(resource, 'name')
+          resource.link = utils.i18n(resource, 'link')
+          if (resource.slug === 'dashboard-tutorial')
+            resource.link = '#'
+          resource.description = utils.i18n(resource, 'description')
+          resource.locked = resource.hidden === 'paid-only' && !store.getters['me/isPaidTeacher']
```

Comment:
I think the test content we have in there might not have had any paid-only resources, but this `isPaidTeacher` getter is not implemented in store/modules/me.coco.js, only store/modules/me.ozar.js, so we'll have to fix that up. For testing, I just marked the [Student Progress Journal](https://codecombat.com/editor/resource/student-progress-journal) as paid-only.